lib/adapters/parsers/grok.js

```diff
@@ -0,0 +1,76 @@
+var base = require('./base');
+var errors = require('../../errors');
+var grok = require('node-grok');
+var Promise = require('bluebird');
+var readlines = require('readlines');
+
+module.exports = base.extend({
+
+    initialize: function(options) {
+        this.pattern = options.pattern || '%{GREEDYDATA:message}';
+    },
+
+    parseStream: function(stream, emit) {
+        var self = this;
+
+        return new Promise(function(resolve, reject) {
+            grok.loadDefault(function(patterns) {
+                var buffer = [];
+                var liner = readlines.readlineStream();
+                var pattern = patterns.createPattern(self.pattern);
+
+                // because of the asynchronous way that patterns parse we need
+                // to track the pending parses and when the stream has ended to
+                // do the right thing.
+                var pending = 0;
+                var streamEnd = false;
+
+                liner.on('readable', function(){
+                    var line;
+                    while ((line = liner.read())) {
+                        line = line.toString();
+                        pending++;
+                        pattern.parse(line, function (err, point) {
+                            if (err) {
+                                reject(errors.runtimeError('RT-INTERNAL-ERROR',{
```

Comment:
I think skip the bad point and log a warning makes the most sense. 

-m


> On Jan 14, 2016, at 6:10 PM, Rodney Lopes Gomes <notifications@github.com> wrote:
> 
> In lib/adapters/parsers/grok.js:
> 
> > +                var pattern = patterns.createPattern(self.pattern);
> > +
> > +                // because of the asynchronous way that patterns parse we need
> > +                // to track the pending parses and when the stream has ended to
> > +                // do the right thing.
> > +                var pending = 0;
> > +                var streamEnd = false;
> > +
> > +                liner.on('readable', function(){
> > +                    var line;
> > +                    while ((line = liner.read())) {
> > +                        line = line.toString();
> > +                        pending++;
> > +                        pattern.parse(line, function (err, point) {
> > +                            if (err) {
> > +                                reject(errors.runtimeError('RT-INTERNAL-ERROR',{
> @demmer so we have two options here:
> 
> reject and then do a stream.close() to stop everything
> trigger a warning like so: https://github.com/juttle/juttle/blob/master/lib/adapters/serializers/csv.js#L25 with the details of the line, skip this line and keep chugging along
> trigger the warning and actually place the line as a point { time: current_time, message: line } into the pipeline ?
> What do you think ?
> 
> —
> Reply to this email directly or view it on GitHub.
>